.github/workflows/ci.yml

```diff
@@ -233,7 +233,7 @@ jobs:
 
       - name: Prepare Docker environment
         run: |
-          echo IMAGE_TAG=$(if [ "$GITHUB_REF" == "refs/heads/main" ]; then echo "main-${GITHUB_SHA::7}"; else echo "$GITHUB_REF_NAME"; fi) >> $GITHUB_ENV
+          echo IMAGE_TAG=$(if [ -n "$GITHUB_REF_NAME" ]; then echo "$GITHUB_REF_NAME"; else echo "main-${GITHUB_SHA::7}"; fi) >> $GITHUB_ENV
```

Comment:
This is only invoked for tags and commits so what you have done looks good to me.